### DISCUSSION TITLE
20241017 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#112] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Bencode2Json

- Added more tests (unit and integration tests).
- Improved docs.
- Peek byte in the main loop without consuming it.
- Capture 1KB from input and output for error context.
- Wrap input and output in [BuffRead](https://doc.rust-lang.org/std/io/trait.BufRead.html) and [BufWrite](https://doc.rust-lang.org/std/io/struct.BufWriter.html) as @da2ce7 suggested.
- Improve errors: added read and write context.
- Added examples
- Moved the repo to Torrust and made it public: https://github.com/torrust/torrust-bencode2json

Next week:

- More examples
- Review crate docs
- Write TEP draft
- Publish a new blog post on [Torrust blog](https://torrust.com/blog)

Next?:

- See [TODO](https://github.com/torrust/torrust-bencode2json?tab=readme-ov-file#todo)
- A repo with a test suite like this: https://github.com/nst/JSONTestSuite for other packages to test the conversion.

Next long-term:

- Json2Bencode?

@da2ce7, @mario-nt  I'd appreciate your feedback, especially on the public API, before the crate's first release (0.1.0). And also about the [Writer trait](https://github.com/torrust/torrust-bencode2json/blob/develop/src/rw/writer.rs#L3-L15).

### Torrust Tracker

There are some new contributors:

- https://github.com/torrust/torrust-tracker/issues/672
- https://github.com/torrust/torrust-tracker/issues/1040

I think we should continue with the plan of [extracting the tracker client into its package](https://github.com/torrust/torrust-tracker/discussions/660).

**Comment by mario-nt** [comment]
### GUI
- https://github.com/torrust/torrust-index-gui/pull/625

### Index

- https://github.com/torrust/torrust-index/pull/742
- https://github.com/torrust/torrust-index/pull/753
